src/ratio/mod.rs

```diff
@@ -0,0 +1,201 @@
+    }
+}
+
+impl ops::Div for Ratio {
+    type Output = Option<Ratio>;
+
+    fn div(self, other: Ratio) -> Option<Ratio> {
+        self.0.checked_div(other.0).map(|total| Ratio(total))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use Ratio;
+
+    #[test]
+    #[should_panic]
+    fn handles_invalid_percentage() {
+        Ratio::from_percentage(101);
+    }
+
+    #[test]
+    #[should_panic]
+    fn handles_invalid_f32() {
+        Ratio::from_f32(1.01);
+    }
+
+    #[test]
+    fn handles_overflow_percentage() {
+        assert_eq!(
+            Ratio::from_percentage(50) + Ratio::from_percentage(55),
+            None
+        );
+        assert_eq!(
+            Ratio::from_percentage(50) - Ratio::from_percentage(55),
+            None
+        );
+        assert_eq!(
+            Ratio::from_percentage(50) * Ratio::from_percentage(55),
+            None
+        );
+    }
+
+    #[test]
+    fn handles_overflow_f32() {
+        assert_eq!(Ratio::from_f32(0.75) + Ratio::from_f32(0.75), None);
+        assert_eq!(Ratio::from_f32(0.25) - Ratio::from_f32(0.75), None);
+        assert_eq!(Ratio::from_f32(0.07) * Ratio::from_f32(0.06), None);
+    }
+
+    #[test]
+    fn adds_percentage() {
+        let a = Ratio::from_percentage(55);
+        let b = Ratio::from_percentage(45);
+        let c = Ratio::from_percentage(10);
+
+        assert_eq!((a + b).unwrap(), Ratio::from_percentage(100));
+        assert_eq!((a + c).unwrap(), Ratio::from_percentage(65));
+    }
+
+    #[test]
+    fn subtracts_percentage() {
+        let a = Ratio::from_percentage(45);
+        let b = Ratio::from_percentage(10);
+        let c = Ratio::from_percentage(1);
+
+        assert_eq!((a - b).unwrap(), Ratio::from_percentage(35));
+        assert_eq!((b - c).unwrap(), Ratio::from_percentage(9));
+    }
+
+    #[test]
+    fn multiplies_percentage() {
+        let a = Ratio::from_percentage(10);
+        let b = Ratio::from_percentage(1);
+        let c = Ratio::from_percentage(2);
+
+        assert_eq!((b * c).unwrap(), Ratio::from_u8(15));
+        assert_eq!((c * c).unwrap(), Ratio::from_u8(25));
+        assert_eq!((a * b).unwrap(), Ratio::from_u8(78));
+    }
+
+    #[test]
+    fn divides_percentage() {
+        let a = Ratio::from_percentage(45);
+        let b = Ratio::from_percentage(10);
+        let c = Ratio::from_percentage(1);
+
+        assert_eq!((b / c).unwrap(), Ratio::from_u8(8));
+        assert_eq!((a / c).unwrap(), Ratio::from_u8(38));
+        assert_eq!((a / b).unwrap(), Ratio::from_u8(4));
+    }
+
+    #[test]
+    fn adds_f32() {
+        let a = Ratio::from_f32(0.55);
+        let b = Ratio::from_f32(0.45);
+        let c = Ratio::from_f32(0.10);
+
+        assert_eq!((a + b).unwrap(), Ratio::from_f32(1.0));
+        assert_eq!((c + c).unwrap(), Ratio::from_u8(52));
+        // This seems to be lossy; possible to test like this?
+        // assert_eq!((c + c).unwrap(), Ratio::from_f32(0.2));
```

Comment:
@chancancode I've added more tests but I think there has to be a better way to test this than what I'm doing right now. The problem seems to be that the conversion is lossy between `f32` to `u8`, and I couldn't find a way to round to a specific decimal point/place value to compare it agains `Ratio::from_f32()`, which is what I'd expect the test to look like.

Reply:
I've addressed all the other comments, so I don't think that this blocks merging, but wanted to note it down.